### PR TITLE
Mobject add_to_back sanity checks

### LINKED ISSUE #1099: Mobjects add_to_back is missing sanity checks
## Description of bug / unexpected behavior
the Mobject.add method contains sanity checks to make sure, no submobject gets added twice and self is not in self.submobjects.
Mobjext.add_to_back is missing this functionality.

## Expected behavior
Adding self should produce a ValueError: Mobject cannot contain self.
Addin a Mobject multiple times should result in it beeing just once in self.submobjects

## How to reproduce the issue
<details><summary>Code for reproducing the problem</summary>

```py
a = Mobject()
b = Line()
a.add_to_back(a)
a.add_to_back(b,b)
print(a.submobjects)
# results in [Mobject, Line, Line]
```

</details>

### PR DESCRIPTION
**Pull request by RickyC0626**
## Motivation

Closes #1099 

## Overview / Explanation for Changes

Additional sanity checks are added to ensure that:
- ValueError is raised when Mobject adds self
- The same Mobject is prevented from being added multiple times (WIP)
- TypeError is raised when a non-Mobject is added

## Testing Status

Specs:
- **CPU**: i5-6400
- **RAM**: 16 GB GDDR3L
- **SSD**: 1 TB SATA
- **GPU**: GTX 1050-Ti 4 GB

WSL: Ubuntu-20.04 running poetry
```py
def test_vmob_add_to_back():
    """Test the Mobject add_to_back method"""
    a = VMobject()
    b = Line()
    with pytest.raises(ValueError):
        # Mobject cannot contain self
        a.add_to_back(a)

    # No submobject gets added twice
    a.add_to_back(b)
    a.add_to_back(b, b)
    assert len(a.submobjects) == 1
```
✔️ Successful test (`test_vectorized_mobject`)

## Further Comments

Duplication prevention may need additional testing for edge cases such as:
```py
a.add_to_back(b, b, b)
a.add_to_back(b, b)
```

## Acknowledgements
- [X] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [X] I have chosen a descriptive PR title (see top of PR template for examples)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The PR title is descriptive enough

### CODE REVIEW

**Comment by GameDungeon** [comment]
It needs black formatting.

File: tests/test_vectorized_mobject.py [hunk 1]
```diff
@@ -85,6 +85,20 @@ def test_vgroup_remove_dunder():
     assert len(obj.submobjects) == 0
 
 
+def test_vmob_add_to_back():
```
Comment (huguesdevimeux) [inline tests/test_vectorized_mobject.py:88]: This test is perfect but why not testing the TypeError thrown [there](https://github.com/ManimCommunity/manim/pull/1194/files#diff-fa8fe466fb7275e64543789145cbd2e35b267715610a6376591ddbda0d2f6194R367) as well ?

**Comment by RickyC0626** [comment]
Since this has been broken beyond repair, I'll just redo it in another PR.
